kolena/_experimental/object_detection/utils.py

```diff
@@ -79,18 +94,29 @@ def _compute_threshold_curve(
             zero_division=0,
         )
 
-        precisions.append(precision)
-        recalls.append(recall)
+        # avoid curves with one x-value and two y-values
+        if recall in recalls:
+            idx = recalls.index(recall)
+            precisions[idx] = max(precisions[idx], precision)
+            recalls[idx] = max(recalls[idx], recall)
```

Comment:
I think ` recalls[idx] = max(recalls[idx], recall)` can be omitted because `recalls[idx] == recall`

Reply:
PR calculations might produce a higher or lower `precision` for the same `recall`. Here, we take the max value to ensure that the highest `precision` of each `recall` is chosen, and not added.

Reply:
Right. That makes sense for `precision` but does it make sense to take the max for `recall`? 

We are taking the index of `recall` so isn't `recalls[idx] == recall` always true?